src/muxed_account.js

```diff
@@ -0,0 +1,155 @@
+ *
+ * Yet would like to use it for 4 different family members:
+ *   1: MA7QYNF7SOWQ3GLR2BGMZEHXAVIRZA4KVWLTJJFC7MGXUA74P7UJUAAAAAAAAAAAAGZFQ
+ *   2: MA7QYNF7SOWQ3GLR2BGMZEHXAVIRZA4KVWLTJJFC7MGXUA74P7UJUAAAAAAAAAAAALIWQ
+ *   3: MA7QYNF7SOWQ3GLR2BGMZEHXAVIRZA4KVWLTJJFC7MGXUA74P7UJUAAAAAAAAAAAAPYHQ
+ *   4: MA7QYNF7SOWQ3GLR2BGMZEHXAVIRZA4KVWLTJJFC7MGXUA74P7UJUAAAAAAAAAAAAQLQQ
+ *
+ * This object makes it easy to create muxed accounts from regular accounts,
+ * duplicate them, get/set the underlying IDs, etc. without mucking around with
+ * the raw XDR.
+ *
+ * Because muxed accounts are purely an off-chain convention, they all share the
+ * sequence number tied to their underlying G... account. Thus, this object
+ * *requires* an {@link Account} instance to be passed in, so that muxed
+ * instances of an account can collectively modify the sequence number whenever
+ * a muxed account is used as the source of a @{link Transaction} with {@link
+ * TransactionBuilder}.
+ *
+ * @constructor
+ *
+ * @param {Account}   account - the @{link Account} instance representing the
+ *                              underlying G... address
+ * @param {string}    id      - a stringified uint64 value that represents the
+ *                              ID of the muxed account
+ *
+ * @see https://developers.stellar.org/docs/glossary/muxed-accounts/
+ */
+export class MuxedAccount {
+  constructor(baseAccount, id) {
+    const accountId = baseAccount.accountId();
+    if (!StrKey.isValidEd25519PublicKey(accountId)) {
+      throw new Error('accountId is invalid');
+    }
+
+    this.account = baseAccount;
+    this._muxedXdr = encodeMuxedAccount(accountId, id);
+    this._mAddress = encodeMuxedAccountToAddress(this._muxedXdr);
+    this._id = id;
+  }
+
+  /**
+   * Parses an M-address into a MuxedAccount object.
+   *
+   * @param  {string} mAddress    - an M-address to transform
+   * @param  {string} sequenceNum - the sequence number of the underlying {@link
+   *     Account}, to use for the underlying base account (@link
+   *     MuxedAccount.baseAccount). If you're using the SDK, you can use
+   *     `server.loadAccount` to fetch this if you don't know it.
+   *
+   * @return {MuxedAccount}
+   */
+  static fromAddress(mAddress, sequenceNum) {
+    const muxedAccount = decodeAddressToMuxedAccount(mAddress);
+    const gAddress = encodeMuxedAccountToAddress(
+      muxedAccount,
+      MUXED_ACCOUNT_FORCE_ED25519
+    );
+    const id = muxedAccount
+      .med25519()
+      .id()
+      .toString();
+
+    return new MuxedAccount(new Account(gAddress, sequenceNum), id);
+  }
+
+  /**
+   * Returns a *new* muxed account with the same address but a new ID
+   *
+   * @param  {string} id    - a stringified uint64 representing the new ID
+   * @return {MuxedAccount} - a new account that shares the underlying base
+   *     account (incl. sequence number) but a different ID set
+   */
+  createSibling(id) {
+    return new MuxedAccount(this.account, id);
+  }
```

Comment:
Thoughts on naming, @leighmcculloch? This is related to #433.

Reply:
Rationale posted in https://github.com/stellar/js-stellar-base/issues/433#issuecomment-1009349938

Reply:
I think we should remove the createSibling function and reconsider it separately if we want to re-add it. It introduces a new concept, sibling, that is one more thing to learn. It is relatively easy to create a muxed account by getting its base account and constructing a new one. Conceptually this seems sounds so I don't think we need createSibling or a function that does what it does. Wdyt?

Reply:
Agreed, done in e9610c077707699e62d6f323ba9063adbc296ef5.